oh2.js

```diff
@@ -123,6 +130,77 @@ function turnOnOff(context, event) {
     rest.postItemCommand(event.payload.accessToken, event.payload.appliance.applianceId, state, success, failure);
 }
 
+function getLockState(context, event) {
+    var success = function (item) {
+        var header = {
+            messageId: event.header.messageId,
+            name: event.header.name.replace("Request", "Response"),
+            namespace: event.header.namespace,
+            payloadVersion: event.header.payloadVersion
+        };
+        var payload = {
+            lockState: item.state == "ON" ? "LOCKED" : "UNLOCKED"
+        };
+        var result = {
+            header: header,
+            payload: payload
+        };
+        utils.log('Done with result', JSON.stringify(result));
+        context.succeed(result);
+    };
+    var failure = function (error) {
+        context.done(null, utils.generateControlError(event.header.messageId, event.header.name, 'DependentServiceUnavailableError', error.message));
+    };
+    rest.getItem(event.payload.accessToken, event.payload.appliance.applianceId, success, failure);
+}
+
+function setLockState(context, event) {
+    var success = function (response) {
+        var header = {
+            messageId: event.header.messageId,
+            name: event.header.name.replace("Request", "Confirmation"),
+            namespace: event.header.namespace,
+            payloadVersion: event.header.payloadVersion
+        };
+        
+        var payload = {};
+        
+        var success = function (item) {
+            payload = {
+                lockState: item.state == "ON" ? "LOCKED" : "UNLOCKED"
+            };
+        };
+        
+        var failiure = function (error) {
+            payload = {
+                lockState: event.payload.lockState == "UNLOCKED" ? "LOCKED" : "UNLOCKED" //reverse to signify error
+            };
+        };
+
+        rest.getItem(event.payload.accessToken, event.payload.appliance.applianceId, success, failiure);
```

Comment:
I think the problem is that no matter what, this will return back the state we just set as a lock will take some time (1 second, 10 seconds, something) and this call will happen milliseconds after we set the state.  Do you agree or do you have other thoughts about how that would work?